one-offs/images-from-boms.go

```diff
@@ -0,0 +1,187 @@
+	s.name = n
+	return s
+}
+
+//type ArtifactSources struct {
+//	DebianRepository   string `yaml:"debianRepository"`
+//	DockerRegistry     string `yaml:"dockerRegistry"`
+//	GitPrefix          string `yaml:"gitPrefix"`
+//	GoogleImageProject string `yaml:"gitPrefix,omitempty"`
+//}
+
+type Image struct {
+	Digest string   `json:"digest"`
+	Tags   []string `json:"tags"`
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+	storageSvc, err := storage.NewClient(ctx, option.WithCredentialsFile(*jsonKeyPath), option.WithScopes(storage.ScopeReadOnly))
+	if err != nil {
+		log.Fatalf("Error generating new storage client: %v", err)
+	}
+
+	boms := make([]*Bom, 0, 100)
+	iter := storageSvc.Bucket(*bucket).Objects(ctx, &storage.Query{Prefix: "bom/"})
+	for obj, err := iter.Next(); err == nil; obj, err = iter.Next() {
+		if releaseRegexp.Match([]byte(obj.Name)) {
+			log.Println(obj.Name)
+			r, err := storageSvc.Bucket(*bucket).Object(obj.Name).NewReader(ctx)
+			if err != nil {
+				log.Fatalf("error reading object %v: %v", obj.Name, err)
+			}
+			bom := &Bom{}
+			if err = yaml.NewDecoder(r).Decode(bom); err != nil {
+				log.Fatalf("error decoding bom %v: %v", obj.Name, err)
+			}
+			boms = append(boms, bom)
+			log.Printf("%+v", bom)
+			//break
+		}
+	}
+
+	// Go get all the images and index them by tag
+	imagesByTag := make(map[string]*Image, 1000)
+	l := boms[0].Services.List()
+	for svcElem := l.Front(); svcElem != nil; svcElem = svcElem.Next() {
+		svc := svcElem.Value.(*Service)
+		log.Printf("Executing gcloud for %v\n", svc.name)
+		cmd := exec.Command("gcloud", "container", "images", "list-tags", fmt.Sprintf("gcr.io/%v/%v", *fromProjID, svc.name), "--format", "json")
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			log.Fatalf("error getting gcloud stdout: %v", err)
+		}
+		if err := cmd.Start(); err != nil {
+			log.Fatalf("error starting: %v", err)
+		}
+
+		images := make([]*Image, 0, 100)
+		if err := json.NewDecoder(stdout).Decode(&images); err != nil {
+			log.Fatalf("error decoding: %v", err)
+		}
+		if err := cmd.Wait(); err != nil {
+			log.Fatalf("error waiting: %v", err)
+		}
+
+		count := 0
+		for _, img := range images {
+			for _, tag := range img.Tags {
+				fullTag := fmt.Sprintf("%v:%v", svc.name, tag)
+				imagesByTag[fullTag] = img
+			}
+			count++
+		}
+		log.Printf("Indexed %v tags\n", count)
+		//break
+	}
+
+	b := strings.Builder{}
+	for _, bom := range boms {
+		l := bom.Services.List()
+		for svcElem := l.Front(); svcElem != nil; svcElem = svcElem.Next() {
+			svc := svcElem.Value.(*Service)
+			fullTag := fmt.Sprintf("%v:%v", svc.name, svc.Version)
+
+			img, ok := imagesByTag[fullTag]
+			if !ok {
+				log.Printf("Can't find %v for bom %v\n", fullTag, bom.Version)
+				continue
+			}
+
+			b.WriteString(fmt.Sprintf("gcloud container images add-tag gcr.io/%v/%v gcr.io/%v/%v:spinnaker-release-%v\n", *fromProjID, fullTag, *toProjId, svc.name, bom.Version))
```

Comment:
See discussion as PR comments, but I think this should be `spinnaker-%v` (ex: `spinnaker-1.19.2`)

Reply:
Done.